packages/schema/src/config/common.ts

```diff
@@ -371,7 +371,9 @@ export default defineUntypedSchema({
    * @type {Array<string | RegExp>}
    */
   watch: {
-    $resolve: val => [].concat(val).filter((b: unknown) => typeof b === 'string' || b instanceof RegExp),
+    $resolve: async (val, get) => [].concat(val).filter((b: unknown) => typeof b === 'string' || b instanceof RegExp).map(
+      async (b: string | RegExp) => typeof b === 'string' ? resolve(await get('srcDir'), b) : b
+    )
```

Comment:
we should await the values _here_ rather than when `options.watch` is used.

```suggestion
    $resolve: async (val, get) => {
      return Promise.all([].concat(val).filter((b: unknown) => typeof b === 'string' || b instanceof RegExp).map(
        async (b: string | RegExp) => typeof b === 'string' ? resolve(await get('srcDir'), b) : b
      ))
	  }
```